tools/opensnoop.py

```diff
@@ -245,13 +269,16 @@
     const char __user *filename = (char *)PT_REGS_PARM2(regs);
     struct open_how __user how;
     int flags;
+    unsigned short mode;
 
     bpf_probe_read_user(&how, sizeof(struct open_how), (struct open_how*)PT_REGS_PARM3(regs));
     flags = how.flags;
+    mode = how.mode;
 #else
 KRETFUNC_PROBE(FNNAME, int dfd, const char __user *filename, struct open_how __user *how, int ret)
 {
     int flags = how->flags;
+    unsigned short mode = how->mode;
```

Comment:
The flag and mode types in struct open_how:
```
struct open_how {
        __u64 flags;
        __u64 mode;
        __u64 resolve;
};
```
Also, should we check O_CREAT | O_TMPFILE flags before saving how->mode?
The same for other open_how usages.